src/AbstractOptions.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Equip\Queue;
+
+use Equip\Command\OptionsInterface;
+use Equip\Command\OptionsSerializerTrait;
+use Equip\Queue\Exception\MessageException;
+
+abstract class AbstractOptions implements OptionsInterface
+{
+    use OptionsSerializerTrait;
+
+    /**
+     * @var string
+     */
+    protected $command;
+
+    /**
+     * @var string
+     */
+    protected $queue;
+
+    /**
+     * Returns the command name
+     *
+     * @return string
+     *
+     * @throws MessageException If the handler property isn't set
+     */
+    public function command()
+    {
+        if (!$this->command) {
+            throw MessageException::missingProperty('command');
```

Comment:
Don't think there's really a way to make it happen earlier without adding a constructor that handles validation.